File: C/Stack/Stack_03.c
/*
===============================================================================
Program Name    : Stack Display Operation
Program Number  : 003
File Name       : Stack03.c

Description     : This program demonstrates Stack using
                  Singly Linear Linked List and supports
                  Push and Display operations.

Date            : August 2026
Language        : C
Author          : Atharv Tushar Bhosale
===============================================================================
*/

#include<stdio.h>
#include<stdlib.h>

struct node
{
    int data;
    struct node *next;
};

typedef struct node NODE;
typedef struct node* PNODE;
typedef struct node** PPNODE;

void Push(PPNODE first,int iNo)
{
    PNODE newn = NULL;

    newn = (PNODE)malloc(sizeof(NODE));

    newn->data = iNo;
    newn->next = *first;

    *first = newn;
}

void Display(PNODE first)
{
    while(first != NULL)
    {
        printf("|%d|\n",first->data);
        first = first->next;
    }
}

int main()
{
    PNODE head = NULL;

    Push(&head,11);
    Push(&head,21);
    Push(&head,51);
    Push(&head,101);

    Display(head);

    return 0;
}

/*
===============================================================================
End of Program
===============================================================================
*/

File: Java/Stack/Stack_03.java
/*
===============================================================================
Program Name    : Stack Display Operation
Program Number  : 003
File Name       : Stack_03.java

Description     : This program demonstrates insertion
                  and display of Stack elements using
                  Singly Linear Linked List.

Concepts Used:
1. Class and Object
2. Linked List
3. Stack Data Structure
4. Dynamic Memory Allocation

Date            : August 2026
Language        : Java
Author          : Atharv Tushar Bhosale
===============================================================================
*/

class node
{
    public int data;
    public node next;
}

class Stack
{
    private node first;
    private int iCount;

    public Stack()
    {
        first = null;
        iCount = 0;
    }

    ///////////////////////////////////////////////////////////////////////////
    //
    // Function Name : Push
    // Description   : Insert element at First
    //
    ///////////////////////////////////////////////////////////////////////////

    public void Push(int iNo)
    {
        node newn = null;

        newn = new node();

        newn.data = iNo;
        newn.next = null;

        newn.next = first;
        first = newn;

        iCount++;
    }

    ///////////////////////////////////////////////////////////////////////////
    //
    // Function Name : Display
    // Description   : Display Stack
    //
    ///////////////////////////////////////////////////////////////////////////

    public void Display()
    {
        node temp = null;

        temp = first;

        System.out.println("\nElements of Stack are : ");

        while(temp != null)
        {
            System.out.println("| " + temp.data + " |");
            temp = temp.next;
        }
    }
}

class Stack_03
{
    public static void main(String A[])
    {
        Stack sobj = new Stack();

        sobj.Push(11);
        sobj.Push(21);
        sobj.Push(51);
        sobj.Push(101);

        sobj.Display();
    }
}

/*
===============================================================================
Sample Output

Elements of Stack are :

| 101 |
| 51 |
| 21 |
| 11 |

===============================================================================
End of Program
===============================================================================
*/
